core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -184,20 +188,52 @@ private[ui] class AllJobsPage(parent: JobsTab, store: AppStatusStore) extends We
     val eventArrayAsStr =
       (jobEventJsonAsStrSeq ++ executorEventJsonAsStrSeq).mkString("[", ",", "]")
 
-    <span class="expand-application-timeline">
-      <span class="expand-application-timeline-arrow arrow-closed"></span>
-      <a data-toggle="tooltip" title={ToolTips.JOB_TIMELINE} data-placement="top">
-        Event Timeline
-      </a>
-    </span> ++
-    <div id="application-timeline" class="collapsed">
-      <div class="control-panel">
-        <div id="application-timeline-zoom-lock">
-          <input type="checkbox"></input>
-          <span>Enable zooming</span>
+    if (totalPages > 0) {
+      <span class="expand-application-timeline">
+        <span class="expand-application-timeline-arrow arrow-closed"></span>
+        <a data-toggle="tooltip" title={ToolTips.JOB_TIMELINE} data-placement="top">
+          Event Timeline
+        </a>
+      </span> ++
+        <div id="application-timeline" class="collapsed">
+          <div class="control-panel">
+            <div id="application-timeline-zoom-lock">
+              <input type="checkbox"></input>
+              <span>Enable zooming</span>
+            </div>
+            <div>
+              <form id={s"form-event-timeline-page"}
+                    method="get"
+                    action=""
+                    class="form-inline justify-content-end"
+                    style="width: 50%; margin-left: auto; margin-bottom: 0px;">
```

Comment:
nit: shall we move the style to webui.css?

Reply:
Thanks. I'll consider it.